docs/source/examplecleanup.rst

```diff
@@ -6,55 +6,94 @@ End-to-End Example
 
-#. Use ``tartufo`` to scan your repository and find any secrets in its history!
 
-   Use what you've learned to scan your repo!
+#. Setup poetry if you want to use the most recent build
+
+   This project uses [Poetry] to manage its dependencies and do a lot of the heavy lifting. So you'll need to clone the tartufo repo and setup poetry!
+
+
+   .. code-block:: console
+
+      git clone git@github.com:godaddy/tartufo.git
+      # use a virtual environment!
+      python3.7 -m venv .venv
+      . .venv/bin/activate
+      
+
+   Next: [Install Poetry]  
+
+   Finally: Run ``poetry install``
+
+
+#. Clone your repo!
+
+   Select and clone the repo you want to run tartufo on
 
    .. code-block:: console
 
-      # Run Tartufo on your repo:
+      # Clone your repo, variables used later:
       GITHUBPROJECT="yourproject"
-      GITHUBREPO="yourrepo.git"
+      GITHUBREPO="myrepo.git"
       GITHUBADDRESS="github.com"
-      tartufo git@${GITHUBADDRESS}:${GITHUBPROJECT}/${GITHUBREPO}
-      # this will spit out a bunch of stuff and end with something like: Results have been saved in /tmp/tmp_jm3hyne
-      TARTUFOSCANPATH="/tmp/tmp_jm3hyne"
-      # this is the number of individual bad password hits tartufo has found:
-      ls ${TARTUFOSCANPATH} | wc -l
+      git clone --mirror git@${GITHUBADDRESS}:${GITHUBPROJECT}/${GITHUBREPO}
 
 
-#. Take results and create a "bad password" file. 
+#. Use ``tartufo`` to scan your repository and find any secrets in its history!
 
-   This file will be used by ``BFG`` and replace these flagged "bad password" entries with ``***REMOVED***``. It is important that you read through this file to make sure there are not exceptions that you want to remove and exclude with tartufo!
+   Use what you've learned to scan your repo!
 
    .. code-block:: console
 
-      # Create a "Bad password" file
-      BADSTRINGARR=`cat ${TARTUFOSCANPATH}/* | jq .strings_found | grep -v '\[' | grep -v '\]'`
-      readarray -t PEWPEWPEWARR <<<"$BADSTRINGARR"
-      # shows entires in array, may be larger than the earlier word count
-      echo ${#PEWPEWPEWARR[@]};
-      SORTED_UNIQUE_ENTRIES=($(echo "${PEWPEWPEWARR[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
-      echo ${#SORTED_UNIQUE_ENTRIES[@]};
-      printf "%s\n" "${SORTED_UNIQUE_ENTRIES[@]}" > badpwz.txt
-      # remove leading/trailing quotes and trailing commas
-      sed -i -e 's/\,$//g' -e 's/\"$//g' -e 's/^\"//g' badpwz.txt
-      
-   Now you have a "bad password" file! Take a look through it, see if anything is wrong, these values will be replaced in your code history
+      # Run Tartufo on your repo and create a list of high entropy items to remove:
+      poetry run tartufo --regex --json --cleanup ${GITHUBREPO} | jq -r '.found_issues[].matched_string' | sort -u > remove.txt
```

Comment:
I don't think that we should be instructing users to use `poetry` here, since that applies only to the development version of `tartufo`, and in fact is not even necessary for that. If the user activates the poetry venv (via `poetry shell`), they can just run `tartufo` directly. But again, we shouldn't be specifying that here. If anywhere, that information would go in the installation and/or contributing docs.

Reply:
Also, why specify `--regex` but not `--entropy`?

Reply:
As the title indicates:
#. Setup poetry if you want to use the most recent build

As this will get merged to master (not stable) and there are still changes in the pipe, it makes sense to have a doc for folks who want to use the codebase in it's current form; ie via poetry. I can add the poetry shell section. 

As for options specified ref: https://github.secureserver.net/CTO/guidelines/blob/master/Standards-Best-Practices/CodeScan/bfg.md

Reply:
> Also, why specify `--regex` but not `--entropy`?

Should be enabled by default, no?

  --entropy / --no-entropy        Enable entropy checks. [default: True]
  --regex / --no-regex            Enable high signal regexes checks. [default:
                                  False]

Reply:
Yeah I suppose there is that. 😄 That works. I just always tend toward explicit. I guess not necessary here though :)